lib/observer.js

```diff
@@ -65,9 +100,28 @@ ObserverMixin.clearObservers = function(operation) {
 
 /**
  * Invoke all async observers for the given operation(s).
+ *
+ * Example:
+ *
+ * Notify all async observers for the `before save` operation.
+ *
+ * ```javascript
+ * var context = {
+    Model: Model,
+    instance: obj,
+    isNewInstance: true,
+    hookState: hookState,
+    options: options,
+ };
+ * Model.notifyObserversOf('before save', context, function(err) {
+    if (err) return cb(err);
+    // user can specify the logic after the observers have been notified
+ });
+ * ```
+ *
  * @param {String|String[]} operation The operation name(s).
  * @param {Object} context Operation-specific context.
- * @param {function(Error=)} callback The callback to call when all observers
+ * @callback {function(Error=)} callback The callback to call when all observers
  *   has finished.
```

Comment:
Oh, cool. Didn't know you could specify the type of the parameter it expects like that.